### PR TITLE
Created PMTCT Dashboard mwana data exporter

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: mwana/apps/export/views.py [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+    
+    site_info_req = OrderedDict()
+
+    try:
+        cursor = connection.cursor()
+        cursor.execute(SQL_QUERY_SELECT) 
+        site_info_req = dictfetchall(cursor)
+
+    except:
+        log = "oops...", sys.exc_info()
```
Comment (tsinkala) [inline mwana/apps/export/views.py:30]: Is this line of code final?

File: mwana/apps/export/views.py [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+from collections import OrderedDict
+
+
+def get_turnaround(request):
+    '''
+    key_value = ""
+    if ((request.method == 'GET') and ('key_value' in request.GET)):
+        key_value = request.GET['key_value']
+    '''
+    connection = psycopg2.connect(host=settings.DATABASES['default']['HOST'], database=settings.DATABASES['default']['NAME'],\
```
Comment (tsinkala) [inline mwana/apps/export/views.py:17]: Any reason for not using connection from django.db, i.e.:

from django.db import connection
cursor = connection.cursor()
...

File: mwana/apps/export/views.py [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+    else:
+        cols = [desc[0] for desc in cursor.description]
+        dict_cols = OrderedDict()
+        for col in cols:
+            dict_cols[col] = ''
+        to_return_dict.append(dict_cols)
+
+    return to_return_dict
+
+SQL_QUERY_SELECT = '''
```
Comment (tsinkala) [inline mwana/apps/export/views.py:60]: Query seems to run fast, but can we use true median? In some cases the differences are quite significant if you use median

Comment (AndreLesa) [inline mwana/apps/export/views.py:60]: Thanks Trevor,

Looping in Paul who wrote the queries.

Comment (AndreLesa) [inline mwana/apps/export/views.py:60]: Hi Trevor,

If you look through the SQL, you’ll observe we used the median for the turnaround as you previously advised and not the mean.

Regards,

Paul C. Kaumba
Software Developer

T +260977664404
M +260966664404
Skype paul.kaumba-palladium
Catholic Relief Services (CRS)

Palladium
Make it possible
thepalladiumgroup.comhttp://thepalladiumgroup.com/

Comment (tsinkala) [inline mwana/apps/export/views.py:60]: Hi Paul,

In the case where you have even number of records, say (11, 22, 33, 44) are
we getting 22 or 33 or (22 + 33)/2?

Thank you,
Trevor

Comment (AndreLesa) [inline mwana/apps/export/views.py:60]: Hi Trevor,

Paul tells me you resolved this when you met?

When can we get a merge?

File: mwana/apps/export/views.py (hunk 2 above)
Comment (tsinkala) [inline mwana/apps/export/views.py:17]: I didn't get feedback on this one

Comment (AndreLesa) [inline mwana/apps/export/views.py:17]: Hi Paul,

Could you clarify on Trevor's final 2 queries?

Trevor, is that a small enough change (the DB connector) that you can fix
as you see fit for the Mwana codebase?

**Comment by AndreLesa** [comment]
Thanks for the work on this Trevor.

I will wait for the accounts.

File: mwana/urls.py [hunk 4]
```diff
@@ -26,6 +26,7 @@
     #url(r"^reports/$", mwana.apps.labresults.views.mwana_reports, name="mwana_reports"),
     #url(r'^$', mwana.apps.labresults.views.mwana_reports, name="mwana_reports"),
     (r'^export/', include('rapidsms.contrib.export.urls')),
+    (r'^exportjson/', include('mwana.apps.export.urls')),
```
Comment (tsinkala) [inline mwana/urls.py:29]: I am moving this line to mwana/zambia/urls.py

**Comment by tsinkala** [comment]
Will create username called dashboard with password dashboardx

**Pull request merged**